miner/src/service_transaction_checker.rs

```diff
@@ -16,21 +16,32 @@
 
 //! A service transactions contract checker.
 
-use call_contract::{CallContract, RegistryInfo};
+use std::collections::HashMap;
+use std::mem;
+use std::sync::Arc;
+use call_contract::{RegistryInfo, CallContract};
 use types::ids::BlockId;
 use types::transaction::SignedTransaction;
 use ethabi::FunctionOutputDecoder;
 use ethereum_types::Address;
+use parking_lot::RwLock;
 
 use_contract!(service_transaction, "res/contracts/service_transaction.json");
 
 const SERVICE_TRANSACTION_CONTRACT_REGISTRY_NAME: &'static str = "service_transaction_checker";
 
 /// Service transactions checker.
-#[derive(Default, Clone)]
-pub struct ServiceTransactionChecker;
+#[derive(Clone)]
```

Comment:
Do we need `Clone`?

Reply:
We need `Clone` in order to share `ServiceTransactionChecker` with `Client` and `PoolClient`.